From 9d2d53ad1136cd17ab5ec94bc969d41d769e870f Mon Sep 17 00:00:00 2001
From: Corey Minyard <cminyard@mvista.com>
Date: Fri, 10 Feb 2017 22:39:16 -0600
Subject: [PATCH] MIPS:kdump: Add some more symbols

Support for 4-level page tables and 48-bit address spaces is coming.
Add symbols so these can be detected and handled properly.

Signed-off-by: Corey Minyard <cminyard@mvista.com>
---
 arch/mips/kernel/machine_kexec.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/mips/kernel/machine_kexec.c b/arch/mips/kernel/machine_kexec.c
index 299ed22..2b17b38 100644
--- a/arch/mips/kernel/machine_kexec.c
+++ b/arch/mips/kernel/machine_kexec.c
@@ -118,7 +118,10 @@ void arch_crash_save_vmcoreinfo(void)
 {
 	VMCOREINFO_NUMBER(PAGE_SHIFT);
 	VMCOREINFO_NUMBER(PGD_ORDER);
-#if defined(CONFIG_32BIT) || !defined(CONFIG_PAGE_SIZE_64KB)
+#if !defined(__PGTABLE_PUD_FOLDED)
+	VMCOREINFO_NUMBER(PUD_ORDER);
+#endif
+#if !defined(__PGTABLE_PMD_FOLDED)
 	VMCOREINFO_NUMBER(PMD_ORDER);
 #endif
 	VMCOREINFO_NUMBER(PTE_ORDER);
@@ -127,6 +130,7 @@ void arch_crash_save_vmcoreinfo(void)
 	VMCOREINFO_NUMBER(_PAGE_HUGE);
 #endif
 #ifdef CONFIG_64BIT
+	VMCOREINFO_NUMBER(TASK_SIZE64);
 #ifdef CONFIG_MAPPED_KERNEL
 	VMCOREINFO_ADDRESS(_text);
 	VMCOREINFO_ADDRESS(_end);
@@ -140,5 +144,6 @@ void arch_crash_save_vmcoreinfo(void)
 
 	VMCOREINFO_ADDRESS(PAGE_OFFSET);
 	VMCOREINFO_ADDRESS(IO_BASE);
+	VMCOREINFO_ADDRESS(MAP_BASE);
 	VMCOREINFO_NUMBER(_PFN_SHIFT);
 }
-- 
2.7.4

